apps/docs/src/.vitepress/utils.ts

```diff
@@ -42,3 +42,97 @@ export const getOnyxNpmPackages = async () => {
 
   return packageFolders;
 };
+
+/**
+ * Extracts the Storybook sidebar folders from all available *.stories.ts files.
+ * Only supports one level of folder nesting, e.g. `components/OnyxHeadline`.
+ * Folders and components will be sorted alphabetically.
+ *
+ * @example
+ * ```ts
+ * {
+ *  components: ["OnyxHeadline", "OnyxButton"],
+ *  utilities: ["GridPlayground"],
+ * }
+ * ```
+ */
+export const getStorybookSidebarFolders = async () => {
+  /**
+   * List of story titles. Can contain folders
+   * @example ["components/OnyxHeadline"]
+   */
+  const storyTitles: string[] = [];
+
+  // extract component folder structure from *.stories.ts files
+  for (const componentName of await getComponents()) {
+    const storybookPath = path.join(COMPONENTS_PATH, componentName, `${componentName}.stories.ts`);
+
+    try {
+      const storybookContent = await fs.readFile(storybookPath, "utf-8");
+      const storyTitle = extractMetaTitleFromStorybook(storybookContent) || componentName;
+      storyTitles.push(storyTitle);
+    } catch {
+      // Storybook file does not exist
+      continue;
+    }
+  }
+
+  const folders = groupComponentsByFolders(storyTitles);
+
+  // sort folders and components alphabetically
+  return Object.keys(folders)
+    .sort()
+    .reduce<typeof folders>((obj, folderName) => {
+      obj[folderName] = folders[folderName].sort();
+      return obj;
+    }, {});
+};
+
+/**
+ * Extracts the "title" property of the Storybook meta object.
+ *
+ * @param fileContent File content of the "*.stories.ts" file.
+ * @example
+ * ```ts
+ * const meta: Meta<typeof OnyxHeadline> = {
+ *   title: "components/OnyxHeadline",
+ * };
+ * // output: "components/OnyxHeadline"
+ * ```
+ */
+const extractMetaTitleFromStorybook = (fileContent: string): string | undefined => {
+  const regexp = /const meta: Meta<typeof .*> = {\n\s*title: "(.*)"/;
+  const matches = regexp.exec(fileContent);
+  return matches?.length === 2 ? matches[1] : undefined;
+};
+
+/**
+ *
+ * Groups the given Storybook meta titles by folder. Only supports one level of nesting.
+ * All titles must have a folder.
+ *
+ * @param storyTitles Storybook meta titles, may contain folders (represented by slashes, e.g. "components/OnyxHeadline")
+ */
+const groupComponentsByFolders = (storyTitles: string[]): Record<string, string[]> => {
+  return storyTitles.reduce<Record<string, string[]>>((groupedFolders, storyTitle) => {
+    // folders, component name will be the last element
+    const folders = storyTitle.split("/");
+
+    if (folders.length > 2) {
+      throw new Error(
+        "Multiple levels of component folders are not supported yet. Please implement it or change the Storybook meta title.",
+      );
+    } else if (folders.length === 1) {
```

Comment:
if `folders.length === 1` is a reason for error, you may throw the error already up here in line 73:

<img width="906" alt="image" src="https://github.com/SchwarzIT/onyx/assets/151019977/8fbf3437-93a2-49ed-a84e-ffbea87ba947">

because falling back on ComponentName will provoke the error here (line 126)?